samples/Client/DaprClient/Program.cs

```diff
@@ -169,6 +172,29 @@ internal static async Task ExecuteStateTransactionAsync(DaprClient client)
             Console.WriteLine("Executed State Transaction!");
         }
 
+         internal static async Task ReadAndSaveStateWithEtagsAsync(DaprClient client)
+        {
+            var origState = new Widget() { Size = "small", Color = "yellow", };
+            await client.SaveStateAsync(storeName, stateKeyName, origState);
+
+            var (origRetrievedState, origEtag) = await client.GetStateAndETagAsync<Widget>(storeName, stateKeyName);
+            Console.WriteLine($"Retrieved state: {origRetrievedState.Size}  {origRetrievedState.Color} with etag: {origEtag}");
+
+            origRetrievedState.Color = "orange";
+            var updatedState = await client.TrySaveStateAsync<Widget>(storeName, stateKeyName, origRetrievedState, origEtag);
+            Console.WriteLine($"Saved modified state with etag: {origEtag} successfully? {updatedState}");
+
+             var (updatedRetrievedState, updatedEtag) = await client.GetStateAndETagAsync<Widget>(storeName, stateKeyName);
+            Console.WriteLine($"Retrieved state: {updatedRetrievedState.Size}  {updatedRetrievedState.Color} with etag: {updatedEtag}");
+
+            updatedRetrievedState.Color = "purple";
+            var isSaveStateSuccess = await client.TrySaveStateAsync<Widget>(storeName, stateKeyName, updatedRetrievedState, origEtag);
```

Comment:
I feel like this needs some comments for explanation.

One of the challenges for sample code is that you want to show how stuff works (and sometimes doesn't) but users will often just copy-paste code and use it verbatim, which is not something you want if you're providing a "bad example".

Can you add some comments that explain that this is intended to *simulate* what happens when you have a concurrency violation.